collector/lib/rust/host/src/info.rs

```diff
@@ -0,0 +1,224 @@
+    }
+
+    /// Whether or not this host machine is Garden Linux
+    pub fn is_garden(&self) -> bool {
+        self.distro().contains("Garden Linux")
+    }
+
+    /// Gets the KernelVersion for this Host
+    /// This must return a Box in order to maintain
+    /// FFI compatibility
+    pub fn kernel_version(&self) -> Box<KernelVersion> {
+        Box::new(self.kernel.clone())
+    }
+
+    /// Gets the ID of this host, populated from /etc/os-release
+    pub fn os_id(&self) -> String {
+        self.os.clone()
+    }
+
+    /// Gets the Build ID of this host, populated from /etc/os-release
+    pub fn build_id(&self) -> String {
+        self.build.clone()
+    }
+
+    /// Gets the distribution of this host, populated from /etc/os-release
+    pub fn distro(&self) -> String {
+        self.distro.clone()
+    }
+
+    /// Gets the hostname of this host, populated either from the
+    /// environment, or from a best-guess look at the filesystem
+    /// (/etc/hostname or /proc/sys/kernel/hostname)
+    pub fn hostname(&self) -> String {
+        self.hostname.clone()
+    }
+
+    pub fn is_rhel76(&self) -> bool {
+        let k = &self.kernel;
+        if self.os_id() == "rhel" || self.os_id() == "centos" && k.release.contains(".el7.") {
+            return (k.kernel == 3 && k.major == 10) && k.build_id >= MIN_RHEL_BUILD_ID;
+        }
+        false
+    }
+
+    pub fn is_rhel86(&self) -> bool {
+        self.os_id() == "rhel" || self.os_id() == "rhcos" && self.kernel.release.contains(".el8_6")
+    }
+
+    pub fn has_ebpf_support(&self) -> bool {
+        self.is_rhel76() || self.kernel.has_ebpf_support()
+    }
+
+    pub fn has_btf_symbols(&self) -> bool {
+        // TODO: not sure I like the layout of this, perhaps it needs a revisit
+        let locations = vec![
+            // try canonical vmlinux BTF through sysfs first
+            ("/sys/kernel/btf/vmlinux".to_string(), false),
+            // fall back to trying to find vmlinux on disk otherwise
+            (format!("/boot/vmlinux-{}", self.kernel.release), false),
+            (
+                format!(
+                    "/lib/modules/{}/vmlinux-{}",
+                    self.kernel.release, self.kernel.release
+                ),
+                false,
+            ),
+            (
+                format!("/lib/modules/{}/build/vmlinux", self.kernel.release),
+                false,
+            ),
+            (
+                format!("/usr/lib/modules/{}/kernel/vmlinux", self.kernel.release),
+                true,
+            ),
+            (
+                format!("/usr/lib/debug/boot/vmlinux-{}", self.kernel.release),
+                true,
+            ),
+            (
+                format!("/usr/lib/debug/boot/vmlinux-{}.debug", self.kernel.release),
+                true,
+            ),
+            (
+                format!("/usr/lib/debug/lib/modules/{}/vmlinux", self.kernel.release),
+                true,
+            ),
+        ];
```

Comment:
```suggestion
        let release = self.kernel.release;
        let locations = vec![
            // try canonical vmlinux BTF through sysfs first
            ("/sys/kernel/btf/vmlinux".to_string(), false),
            // fall back to trying to find vmlinux on disk otherwise
            (format!("/boot/vmlinux-{release}"), false),
            (format!("/lib/modules/{release}/vmlinux-{release}"), false),
            (format!("/lib/modules/{release}/build/vmlinux"), false),
            (format!("/usr/lib/modules/{release}/kernel/vmlinux"), true),
            (format!("/usr/lib/debug/boot/vmlinux-{release}"), true,),
            (format!("/usr/lib/debug/boot/vmlinux-{release}.debug"), true),
            (format!("/usr/lib/debug/lib/modules/{release}/vmlinux"), true),
        ];
```

Reply:
You could also directly call `host_path` on the ones that need to be mapped and drop from a vector of tuples to a vector of paths.